dspace-api/src/main/java/org/dspace/core/Context.java

```diff
@@ -641,6 +658,53 @@ public long getCacheSize() throws SQLException {
     }
 
     /**
+     * Change the mode of this current context.
+     *
+     * BATCH_EDIT: Enabling batch edit mode means that the database connection is configured so that it is optimized to
+     * process a large number of records.
+     *
+     * READ_ONLY: READ ONLY mode will tell the database we are nog going to do any updates. This means it can disable
+     * optimalisations for delaying or grouping updates.
+     *
+     * READ_WRITE: This is the default mode and enables the normal database behaviour. This behaviour is optimal for querying and updating a
+     * small number of records.
+     *
+     * @param newMode The mode to put this context in
+     */
+    public void setMode(Mode newMode) {
+        try {
+            //update the database settings
+            switch (newMode) {
+                case BATCH_EDIT:
+                    dbConnection.setConnectionMode(true, false);
+                    break;
+                case READ_ONLY:
```

Comment:
That's a good and valid point. I think the easiest way is to clear the cache when switching to READ_ONLY. I'll try to add that to this PR.